src/components/AdminPanelLayout/AdminPanelLayout.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import {Layout} from '../../pages/Layout/Layout';
+import {Col, ListGroup, Row, TabContainer, TabContent, TabPane} from 'react-bootstrap';
+import {StatisticsPane} from './StatisticsPane';
+import {ClusterPane} from './ClusterPane';
+import UsersPane from './UsersPane';
+import {DowntimeScheduler} from './DowntimeScheduler';
+
+interface AdminPanelLayoutProps {
+  defaultActivePanel?: '#statistics'|'#cluster-management'|'#user-management';
+}
+
+const panes = [
+  {label: 'Application Statistics', eventKey: '#statistics', component: <StatisticsPane/>},
+  {label: 'Cluster Management', eventKey: '#cluster-management', component: <ClusterPane/>},
+  {label: 'User Management', eventKey: '#user-management', component: <UsersPane/>},
+  {label: 'Downtime Scheduler', eventKey: '#downtime-scheduler', component: <DowntimeScheduler/>}
+];
+
+export const AdminPanelLayout = (props: AdminPanelLayoutProps) => (
+  <Layout>
+    <h1>Admin Console</h1>
+    <TabContainer defaultActiveKey={(props.defaultActivePanel) ? props.defaultActivePanel : '#statistics'}>
```

Comment:
You are duplicating the eventKey possibilities in the defaultActivePanel type definition.

Go ahead and make an alias of the types by marking the array as a constant array and generate an alias for the items of the array.


When passing the `defaultActiveKey` you can call the cast helper function to ensure that `#statistics` exists as a tab or a compile error will show.

```suggestion
interface AdminPanelLayoutProps {
  defaultActivePanel?: AdminTabKeys;
}

const panes = [
  {label: 'Application Statistics', eventKey: '#statistics', component: <StatisticsPane/>},
  {label: 'Cluster Management', eventKey: '#cluster-management', component: <ClusterPane/>},
  {label: 'User Management', eventKey: '#user-management', component: <UsersPane/>},
  {label: 'Downtime Scheduler', eventKey: '#downtime-scheduler', component: <DowntimeScheduler/>}
] as const;

type AdminTabKeys = typeof panes[number]['eventKey'];

export const AdminPanelLayout = (props: AdminPanelLayoutProps) => (
  <Layout>
    <h1>Admin Console</h1>
    <TabContainer defaultActiveKey={cast<AdminTabKeys>(props.defaultActivePanel ? props.defaultActivePanel : '#statistics')}>
```

Reply:
Pushed these changes as you suggested